include/albatross/src/graph/minimum_spanning_tree.hpp

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2021 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef ALBATROSS_GRAPH_MINIMUM_SPANNING_TREE_HPP_
+#define ALBATROSS_GRAPH_MINIMUM_SPANNING_TREE_HPP_
+
+namespace albatross {
+
+template <typename VertexType> struct Edge {
+
+  Edge(VertexType a_, VertexType b_, double cost_, std::size_t id_ = 0)
+      : a(a_), b(b_), cost(cost_), id(id_){};
+
+  bool operator<(const Edge &other) const {
+    return std::tie(this->cost, this->a, this->b) <
+           std::tie(other.cost, other.a, other.b);
+  }
+
+  bool operator==(const Edge &other) const {
+    return std::tie(this->cost, this->a, this->b) ==
+           std::tie(other.cost, other.a, other.b);
+  }
+
+  friend std::ostream &operator<<(std::ostream &os,
+                                  const Edge<VertexType> &edge) {
+    os << edge.a << " <--> " << edge.b << " [" << edge.cost << "]";
+    return os;
+  }
+
+  VertexType a;
+  VertexType b;
+  double cost;
+  std::size_t id;
```

Comment:
Is this member useful/necessary? Don't see it utilized anywhere. Ostensibly it might be used as a tie-breaker in the equality comparison, but that doesn't seem to be the case.

Reply:
Yeah I left this in here to make it easy to inverse a conversion from some problem to a spanning tree problem.  For example, say you have some graph defined by some more complicated types:
```
std::vector<Foo> foos = get_application_specific_data();
```
You can build an `albatross::Graph` from them:
```
std::vector<albatross::Edge> edges;
for (std::size_t i = 0; i < foos.size(); ++i) {
  edges.push_back(Edge(foo.a, foo.b, cost(foo), i);
};
```
Find the minimum spanning tree,
```
const auto solution = minimum_spanning_tree(create_graph(edges));
```
And then convert back to the problem specific types
```
std::vector<Foo> foo_solution;
for (const auto &edge : solution.edges) {
  foo_solution.push_back(foos[edge.id]);
}
```
This is actually how I'm using this code in our parent repos.